code/datums/components/sign_language.dm

```diff
@@ -196,6 +198,29 @@
 
 	return SIGN_OKAY
 
+/**
+ * Check if we can sign the given spell
+ *
+ * Checks to make sure the spell is not a mime spell, and that we are able to physically cast the spell.
+ * Arguments:
+ * * mob/living/carbon/source - the caster of the spell
+ * * datum/action/cooldown/spell/spell - the spell we are trying to cast
+ * * feedback - whether or not a message should be displayed in chat
+ * *
+ * * returns SPELL_INVOCATION_FAIL or SPELL_INVOCATION_SUCCESS
+ */
+/datum/component/sign_language/proc/can_cast_spell(mob/living/carbon/source, datum/action/cooldown/spell/spell, feedback)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/carbon_parent = parent
+	if(spell.invocation_type == INVOCATION_EMOTE) // Mime spells are not cast with signs
+		return
```

Comment:
This is not returning success or fail

Reply:
Vinylspiders is correct. This isn't intended to return success or fail, as mime spells (or any emote invocation spells for that matter, though I'm fairly certain this exists only for mime spells) aren't meant to go through this check at all. Without this, it means that if a mute sign language player chooses to play mime, their spells will also be forced to go through the sign language check despite not being spoken.